src/main/scala/io/iohk/ethereum/domain/Account.scala

```diff
@@ -42,6 +42,9 @@ case class Account(
   def withStorage(storageRoot: ByteString): Account =
     copy(storageRoot = storageRoot)
 
+  def resetAccount(startNonce: UInt256 = UInt256.Zero): Account =
+    copy(nonce = startNonce, storageRoot = Account.EmptyStorageRootHash, codeHash = Account.EmptyCodeHash)
```

Comment:
If we really need this method then how about `resetAccountPreservingBalance`? I'm not fond of long names but I feel `reset` may still be misleading.